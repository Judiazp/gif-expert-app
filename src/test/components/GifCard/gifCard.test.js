import React from 'react';
import '@testing-library/dom';
import { shallow } from 'enzyme';
import GifCard from '../../../components/gifCard/index'

describe('Pruebas en el componente GifCard', () => {

    const title = 'title'
    const url = 'https://urlGif.com'
    const wrapper = shallow( <GifCard title={ title } url={ url } /> )

    test('Debe mostrar el componente GifCard', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe tener un h4 con el title', () => {
        const h4 = wrapper.find('h4');
        expect( h4.text().trim() ).toBe( title );
    })

    test('la imagen debe tener el atributo src igual a la url y alt igual al titulo', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('debe tener la clase animate__fadeInDown', () => {
        const div = wrapper.find('div');
        const className = div.prop('className')
        expect(className.includes('animate__fadeInDown')).toBe( true )
    });
})