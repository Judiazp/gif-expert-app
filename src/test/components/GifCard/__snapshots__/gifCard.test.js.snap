// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifCard Debe mostrar el componente GifCard 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="title"
    src="https://urlGif.com"
  />
  <h4
    className="gif-title"
  >
     
    title
     
  </h4>
</div>
`;
